src/Microsoft.ML.Data/Utilities/LocalEnvironment.cs

```diff
@@ -6,13 +6,16 @@
 
 namespace Microsoft.ML.Runtime.Data
 {
+    using System.ComponentModel.Composition.Hosting;
     using Stopwatch = System.Diagnostics.Stopwatch;
 
     /// <summary>
     /// An ML.NET environment for local execution.
     /// </summary>
     public sealed class LocalEnvironment : HostEnvironmentBase<LocalEnvironment>
     {
+        private readonly Func<CompositionContainer> _compositionContainerFactory;
```

Comment:
>_compositionContainerFactory [](start = 52, length = 28)

yet another reason to make `LocalEnvironment` and `MLContext` one thing

Reply:
This was I thought the plan, but I could be wrong, did we change our minds on this?

---
In reply to: [228673857](https://github.com/dotnet/machinelearning/pull/1406#discussion_r228673857) [](ancestors = 228673857)

Reply:
Hmm, does this mean that `HostEnvironmentBase` is to implement `IHostEnvironment` explicitly? 
In any case, this might have been your plan originally, but I didn't think of it back then, and I would rather not do it right now in this particular PR.

---
In reply to: [229093529](https://github.com/dotnet/machinelearning/pull/1406#discussion_r229093529) [](ancestors = 229093529,228673857)

Reply:
Yes, I believe that is the implication. I think I even brought it up as a natural consequence, but maybe I forgot. Certainly this is not the PR, I agree. Just clarifying the status of our long range plans.

---
In reply to: [229394800](https://github.com/dotnet/machinelearning/pull/1406#discussion_r229394800) [](ancestors = 229394800,229093529,228673857)